### PR TITLE
add a crude CLI tool to unset jail properties

### PR DESCRIPTION
**Pull request by igalic**
so far it's a copy of `get` with some fixups

### CODE REVIEW

File: ioc_cli/unset.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    key: str
+) -> None:
+    del resource.config[key]
+
+
+def _delete_jail_value(
+    resource: 'libioc.LaunchableResource.LaunchableResource',
+    key: str
+) -> None:
+    del resource.config[key]
```
Comment (igalic) [inline ioc_cli/unset.py]: so far, this does nothing.
Is there anything else i need to do to persist a deleted setting?

File: ioc_cli/unset.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            exit(1)
+
+    if (_prop is None) and (jail == ""):
+        logger.error("Missing arguments property and jail")
+        exit(1)
+    elif (_prop is not None) and (jail == ""):
+        logger.error("Missing argument property name")
+        exit(1)
+
+    if _prop:
```
Comment (igalic) [inline ioc_cli/unset.py]: this needs a loop.

File: ioc_cli/unset.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        try:
+            delete_method(source_resource, _prop)
+        except KeyError:
+            logger.error(f"Unknown property '{_prop}'")
+            exit(1)
+
+        print(_prop)
+
```
Comment (igalic) [inline ioc_cli/unset.py]: and after the loop, we'll need a `source_resource.save()`

**Comment by igalic** [comment]
hrm… not yet…

```sh-session
root@container-host1 /u/l/s/ioc# ioc unset basejail_type webirc
Deleting property 'basejail_type'
Unknown property 'basejail_type'
root@container-host1 /u/l/s/ioc# cat /iocage/jails/webirc/config.json
```
```json
{
    "basejail": "yes",
    "basejail_type": "nullfs",
    "depends": "dhcp",
    "id": "webirc",
    "release": "12.0-RELEASE",
    "user.facts.role": "webirc",
    "user.pkglist": "puppet6,git-lite,node8,npm-node8,python27"
}
```

File: ioc_cli/unset.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        logger.error("Missing arguments property and jail")
+        exit(1)
+    elif (_prop is not None) and (jail == ""):
+        logger.error("Missing argument property name")
+        exit(1)
+
+    if _prop:
+        for p in _prop:
+            try:
+                del source_resource.config[_prop]
```
Comment (igalic) [inline ioc_cli/unset.py]: this should be `p` not `_prop`.

File: ioc_cli/unset.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    if _prop:
+        for p in _prop:
+            try:
+                del source_resource.config[p]
+            except KeyError:
+                logger.error(f"Unknown property '{p}'")
+                exit(1)
+
+            print(f"Deleting property '{p}'")
```
Comment (igalic) [inline ioc_cli/unset.py]: we should make this output the same as `set`

**Comment by gronke** [comment]
Jail config properties can already be unset by leaving them empty. There is no need to add an additional command.

### Set a Jail config value to None
Both commands achieve the same goal, setting `ip4_addr` to None:
```sh
ioc set ip4_addr= <MYJAIL>
ioc set ip4_addr=- <MYJAIL>
```

### Delete a Jail config property
Whatever value was set, the property is removed from the config(file) entirely. When modifying defaults, ioc will fall back to the hardcoded global defaults again:
```sh
ioc set ip4_addr <MYJAIL>
```

**Comment by igalic** [comment]
so i guess my issue was that i was trying to unset a virtual property

**Pull request closed without merging**